src/item.h

```diff
@@ -493,6 +510,23 @@ class item : public JsonSerializer, public JsonDeserializer
  mtype*   corpse;
  std::vector<item> contents;
 
+protected:
+        // amount of storage used by bag
+        int storage_occupied;
```

Comment:
This looks like a simple cache of the total contained volume. It can be recalculated when needed for the few items that require it instead of storing it for *every* item in existence.